packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -99,32 +103,18 @@ async function instantiateModule(
   const isCircular = (url: string) => urlStack.includes(url)
 
   const {
-    isProduction,
-    resolve: { dedupe, preserveSymlinks },
+    resolve: { dedupe },
     root
   } = server.config
 
-  // The `extensions` and `mainFields` options are used to ensure that
-  // CommonJS modules are preferred. We want to avoid ESM->ESM imports
-  // whenever possible, because `hookNodeResolve` can't intercept them.
-  const resolveOptions: InternalResolveOptions = {
-    dedupe,
-    extensions: ['.js', '.cjs', '.json'],
-    isBuild: true,
-    isProduction,
-    isRequire: true,
-    mainFields: ['main'],
-    preserveSymlinks,
-    root
-  }
-
   // Since dynamic imports can happen in parallel, we need to
   // account for multiple pending deps and duplicate imports.
   const pendingDeps: string[] = []
 
   const ssrImport = async (dep: string) => {
     if (dep[0] !== '.' && dep[0] !== '/') {
-      return nodeImport(dep, mod.file!, resolveOptions)
+      const [, resolvedId] = await moduleGraph.resolveUrl('/@id/' + dep, true)
```

Comment:
Unsure if Node.js built-ins are properly handled by this line

Reply:
there is a constant for /@id/ somewhere in utils  and wouldn't the moduleGraph only know about modules already imported?

Reply:
> wouldn't the moduleGraph only know about modules already imported?

Nope, the `resolveUrl` method is just a wrapper around `pluginContainer.resolveId`.